tools/run_tests/run_xds_tests.py

```diff
@@ -1004,6 +1004,10 @@ def test_path_matching(gcp, original_backend_service, instance_group,
                 if compare_expected_instances(stats, expected_instances):
                     logger.info("success")
                     break
+                elif i == retry_count - 1:
```

Comment:
Use return instead of break above, and then you can raise unconditionally after the for loop exits? That would avoid this if statement and make it cleared what triggers the failure

Reply:
But return would also break the outer loop, and skip the rest test cases. We only want to break the inner loop.